### ISSUE TITLE
image quality not working

### ISSUE THREAD

**Issue by tsonevn** [NativeScript/NativeScript#4404]
_From @slasho on June 16, 2017 3:57_

reproducing this example, on line 105, I added the quality option in 50 and the image saved is 2MB grater than the original...  if I changed to 100 or 1 same result.

https://github.com/NativeScript/sample-ImageUpload/blob/master/app/main-page.js#L105

_Copied from original issue: NativeScript/nativescript-imagepicker#95_

**Comment by tsonevn** [comment]
Hi @slasho,
Could you provide some more info about the problem, which you are facing with? It would also help if you give us info about the image(its format and size).

**Comment by slasho** [comment]
Hi @tsonevn , In my app when use nativescript-imagepicker  with the code below
the **saveToFile** quality parameter is not working, because I select an 1MB jpg image and it supposed to save it with 50 of quality, hoping to get a smaller image, but instead I get a 3MB image.


Using:
```
┌──────────────────┬─────────────────┬────────────────┬──────────────────┐
│ Component        │ Current version │ Latest version │ Information      │
│ nativescript     │ 2.5.4           │ 3.0.3          │ Update available │
│ tns-core-modules │ 2.5.2           │ 3.0.1          │ Update available │
│ tns-android      │ 3.0.1           │ 3.0.1          │ Up to date       │
│ tns-ios          │ 3.0.1           │ 3.0.1          │ Up to date       │
└──────────────────┴─────────────────┴────────────────┴──────────────────┘
```

`function startSelection(context) {
	context
		.authorize()
		.then(function() {
            imageItems.length = 0;
			return context.present();
		})
		.then(function(selection) {
			selection.forEach(function(selected_item) {
                selected_item.getImage().then(function(imagesource){
                    let folder = fs.knownFolders.documents();
                    let path = fs.path.join(folder.path, "Test"+counter+".jpg");
                    **let saved = imagesource.saveToFile(path, "jpg",50);**
                    
                    if(saved){
                        // OK
                    }
                    counter++;
                })
                
			});
		}).catch(function (e) {
			console.log(e.eventName);
		});
}`

**Comment by slasho** [comment]
@tsonevn hello, any ideas on this?

**Comment by tsonevn** [comment]
Hi @slasho,
I investigate further this problem and found that `quality` property will not work as expected on Android.
Depending on the format, the image will be saved with the same size for `png` or it will be saved with greater size for those in `JPG` format
I am attaching the project, where this problem could be reproduced.
For further info and a possible fix, please keep track the issue.

[Archive.zip](https://github.com/NativeScript/NativeScript/files/1088039/Archive.zip)

**Comment by slasho** [comment]
@tsonevn 👍 I'll have to wait until the fix

Thanks

**Comment by tsonevn** [comment]
Hi @slasho,
I investigate further this issue and found that the issue is related to setting the wrong format for the `Bitmap.CompressFormat`, while setting `JPG` as a format in the `saveToFile` method.  This has been already fixed with PR [#4544](https://github.com/NativeScript/NativeScript/pull/4544) and it will be available for some of the next version of the modules. 
However, in the meantime, you could set `jpeg` as a format, which will allow you to use the quality option for the `image-source`. For example:
```
saveToFile(path, "jpeg",50);
```
